src/duplicacy_b2client.go

```diff
@@ -144,15 +159,24 @@ func (client *B2Client) call(url string, input interface{}) (io.ReadCloser, int6
 		response.Body.Close()
 		if response.StatusCode == 401 {
 			if url == B2AuthorizationURL {
-				return nil, 0, fmt.Errorf("Authorization failure")
+				return nil, nil, 0, fmt.Errorf("Authorization failure")
 			}
 			client.AuthorizeAccount()
 			continue
 		} else if response.StatusCode == 403 {
 			if !client.TestMode {
-				return nil, 0, fmt.Errorf("B2 cap exceeded")
+				return nil, nil, 0, fmt.Errorf("B2 cap exceeded")
 			}
 			continue
+		} else if response.StatusCode == 404 {
+			if http.MethodHead == method {
+				return nil, nil, 0, fmt.Errorf("URL request '%s' returned status code %d", url, response.StatusCode)
```

Comment:
Can't return an error here.  Should instead tell the caller that the given file doesn't exist in the storage.

Reply:
I'm not sure I follow. If I look at how the code is used, returning an error does indicate that the file doesn't exist.

b2storage.GetFileInfo for example

Are you saying it should just be swallowing the error and returning "nil, nil, 0, nil"?

Reply:
Yes, if the file to be looked up doesn't exist, GetFileInfo expects ListFileNames to return an empty Entry array, so maybe here should return a responseHeader of nil and in ListFileNames use that as the condition to return the empty Entry array.

Reply:
I've updated the 404 branch to return a nil error and then ListFileNames treats that nil responseHeader as a missing file and returns an empty array.